spec/array_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'array'
+
+describe Array do
+  subject {[1,2,3,4]}
+  describe 'faisal_inject' do
+
+    it 'adds' do
+      expect(subject.faisal_inject{|k,v| k + v}).to eq 10
+    end
+
+    it 'adds with an argument' do
+      expect(subject.faisal_inject(10){|k,v| k + v}).to eq 20
+    end
+
+    it 'adds with an argument' do
+      expect(subject.faisal_inject(10){|k,v| k * v}).to eq 240
+    end
+
+    it 'takes in an addition symbol' do
+      expect(subject.faisal_inject(:+)).to eq 10
+    end
+
+    it 'takes in an subtraction symbol' do
+      expect(subject.faisal_inject(:-)).to eq -8
+    end
+
+    it 'takes in an division symbol' do
+      expect(subject.faisal_inject(:/)).to eq 0
+    end
+
+    it 'takes in an multiplication symbol' do
+      expect(subject.faisal_inject(:*)).to eq 24
+    end
+
+    it 'can take a 2 arguments (-)' do
+      expect(subject.faisal_inject(10,:+)).to eq 20
+    end
+
+    it 'can take a 2 arguments (+)' do
+      expect(subject.faisal_inject(10,:*)).to eq 240
+    end
+
+    it 'works with hashes' do
+      expect(subject.faisal_inject({}){|k,v| k[v] = 3; k}).to eq ({1=>3, 2=>3, 3=>3, 4=>3})
```

Comment:
Line is too long. [91/80]<br>Space missing after comma.<br>Surrounding space missing for operator '=>'.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.<br>Space inside { missing.<br>Space inside } missing.